scripts/comfyui.py

```diff
@@ -57,21 +59,47 @@ def postprocess_batch_list(self, p, pp, *args, **kwargs):
         if len(pp.images) == 0:
             return
 
-        batch_results = external_code.run_workflow(
-            workflow_type=default_workflow_types.postprocess_workflow_type,
-            tab=self.get_tab(),
-            batch_input=list(pp.images),
-        )
+        all_results = []
+        for batch_input in extract_contiguous_buckets(pp.images, p.batch_size):

```

Comment:
I don't understand why we are now running N workflows instead of the initial single one we were running previously, as well as why the `extract_contiguous_buckets` is usefull here.

Reply:
You will run a single workflow as long as we are the first extension to return images in `batch_postprocess_list`. The problem is that other extensions can also return images arbitrarily here, so we need to make sure we can actually call `torch.stack()` on the batch. To do this, I take the longest sequence of inputs in `pp.images` that effectively have the same shape, but not longer than the webui batch size so that comfyui doesn't get an enourmous batch size if another extension doubled, tripled+ the number of images in `pp.images`.

Reply:
What I get from what you're saying is that each "contiguous_bucket" is smaller or equal in size to the batch size? Why not make it equal? Or did I misundertand what you meant?

Reply:
Also, why can't other extensions handle their own run_workflow code in their own postprocess_batch_list function?